node/src/tests/common.rs

```diff
@@ -0,0 +1,30 @@
+use tokio::sync::mpsc;
+
+use crate::indexer::{
+    tx_sender::{TransactionProcessorError, TransactionSender, TransactionStatus},
+    types::ChainSendTransactionRequest,
+};
+
+#[derive(Debug, Clone)]
+pub struct MockTransactionSender {
+    pub transaction_sender: mpsc::Sender<ChainSendTransactionRequest>,
+}
+
```

Comment:
I'm a bit torn on whether this struct provides any value right now or if we should just use `TransactionProcessorHandle` directly in our tests. I guess this one makes it easier to reuse the existing code with smaller modifications for tests. But then again, why don't we just implement `TransactionSender` for `mpsc::SenderzChainSendTransactionRequests>` directly?